return empty accesslist rlp

JeeChoi opened the pull request:
This is my attempt at fixing a nil pointer deference bug.

I ran across this bug when I tried to create a tx of type `TransactionTypeDynamicFee` without filling in an accessList and then sign that tx. 

The crash happens on this line: https://github.com/INFURA/go-ethlibs/blob/e0d82d62fa8f55cd0ef5b7804fbfe61e1255ccd8/eth/transaction_signing.go#L155

I was using the other unit-tests as references but I wasn't sure how some of the values were filled out so left them as a TODO. Did you use another library to sign/send the tx to get the txhash, signed values etc?

ryanschneider commented:
@JeeChoi we can cheat a little now that Ropsten is forked by crafting a tx for it.  I updated the tests in https://github.com/INFURA/go-ethlibs/pull/53/commits/829ee7ce2ef48286ffa65472f007ac5fd2f70d32 by doing the following:

- switched the chainId to ropsten 0x3
- ran the test, and replaced the generated values for `expectedUnsigned` and `expectedSigned`

I then passed the expectedSigned value into a ropsten geth console and got:

```
> eth.sendRawTransaction("0x02f86a038085039b2eb2008507365d64008094df0a88b2b68c673713a8ec826003676f272e35730180c080a0f0019f2823699d9c29de7da61088f020dff2014bc542d25082715081cce4d64aa01ee67c1cc8c4063e5cf3d9fbab8abf42a1f653ee41725786365f74784c8e213b")
Error: nonce too low
	at web3.js:6357:37(47)
	at web3.js:5091:62(37)
	at <eval>:1:23(4)
```

Since this correctly returned `nonce too low` it means the TX was otherwise accepted as valid, so then proceeded to flesh out the test with the correct values for the hash, R, S, and V.

Alternatively we could use the private key on a local node and use `eth.FillTransaction()` which is what I did for the previous EIP-2930 style txs, but the net result would be the same.

JeeChoi commented:
> @JeeChoi we can cheat a little now that Ropsten is forked by crafting a tx for it. I updated the tests in [829ee7c](https://github.com/INFURA/go-ethlibs/commit/829ee7ce2ef48286ffa65472f007ac5fd2f70d32) by doing the following:
> 
> * switched the chainId to ropsten 0x3
> * ran the test, and replaced the generated values for `expectedUnsigned` and `expectedSigned`
> 
> I then passed the expectedSigned value into a ropsten geth console and got:
> 
> ```
> > eth.sendRawTransaction("0x02f86a038085039b2eb2008507365d64008094df0a88b2b68c673713a8ec826003676f272e35730180c080a0f0019f2823699d9c29de7da61088f020dff2014bc542d25082715081cce4d64aa01ee67c1cc8c4063e5cf3d9fbab8abf42a1f653ee41725786365f74784c8e213b")
> Error: nonce too low
> 	at web3.js:6357:37(47)
> 	at web3.js:5091:62(37)
> 	at <eval>:1:23(4)
> ```
> 
> Since this correctly returned `nonce too low` it means the TX was otherwise accepted as valid, so then proceeded to flesh out the test with the correct values for the hash, R, S, and V.
> 
> Alternatively we could use the private key on a local node and use `eth.FillTransaction()` which is what I did for the previous EIP-2930 style txs, but the net result would be the same.

Thanks Ryan, will keep in mind for next time.
